kernel/os/include/os/arch/pic32/os/os_arch.h

```diff
@@ -38,7 +38,8 @@ typedef uint32_t os_sr_t;
 /* Stack type, aligned to a 32-bit word. */
 #define OS_STACK_PATTERN    (0xdeadbeef)
 
-typedef uint32_t os_stack_t;
+/* uint64_t in an attempt to get the stack 8 aligned */
+typedef uint64_t os_stack_t;
```

Comment:
When running on sim x86 we have similar restrictions, i.e. stack has to be aligned to 16 bytes.
The way this was dealt with was by making sure at task's stack init function that the SP was aligned to that boundary.
If you can follow the assembly: kernel/os/src/arch/sim/os_arch_stack_frame.s
Although the code is commented :)